main/auth/google.py

```diff
@@ -55,17 +55,14 @@ def retrieve_user_from_google(response):
 
   if 'email' in response:
     email = response['email']
-  elif 'emails' in response:
-    email = response['emails'][0]['value']
   else:
     email = ''
 
-  if 'displayName' in response:
-    name = response['displayName']
-  elif 'name' in response:
-    names = response['name']
-    given_name = names.get('givenName', '')
-    family_name = names.get('familyName', '')
+  if 'name' in response:
+    name = response['name']
+  elif 'given_name' in response and 'family_name' in response:
```

Comment:
This "elif" is not needed. Note that with the @gmist suggested and accepted change, the code (if this "elif" is removed) will simply try to get the "given_name" and "family_name" (or get empty strings); after which the "join"+"strip" will try its best to derive a "name".

At the moment, the "elif" (suggested to be dropped) will prevent that unless both "given_name" and "family_name" are present... and that was not what the old code did and not what I expect we should require.

If both are not present, the "elif" will be skipped and the "else" will then create a 'google_user_'-based "name".

I think you should replace that "else" with an ```if not name:``` so that any (however achieved) empty name will be avoided by assigning a 'google_user_'-based "name".

Reply:
In fact, the current ```elif 'given_name' in response and 'family_name' in response:``` should become the ```else:``` here; as the current ```else:``` below is suggested to be changed to ```if not name:``` already.